pkg/controller/reconciler/reconciler.go

```diff
@@ -0,0 +1,91 @@
+package reconciler
+
+import (
+	"context"
+	"fmt"
+
+	"go.mongodb.org/atlas-sdk/v20231115008/admin"
+	"go.uber.org/zap"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/translation/project"
+
+	akov2 "github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1"
+)
+
+type Reconciler struct {
```

Comment:
**question**: the type implements project specific reconciliation bits, that is resolving atlas projects via kube or atlas and resolving credentials for a project however the type is very abstract named "Reconciler" and clashes with controller-runtime concepts. I am much rather inclined to name it concretely:

```suggestion
type AtlasProjectReconciler struct {
```

Reply:
uhm but ti is not an atlas project reconciler 

What about `AtlasReconciler` instead?

Reply:
much better 👍

Reply:
renamed like that